### PR TITLE
Update compliance tasks in python packaging pipeline and fix some compile warnings

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Update SDLNativeRules from v2 to v3. The new one allows us setting excluded paths.
2. Update TSAUpload from v1 to v2. And add a config file ".gdn/.gdntsa" for it.
3. Fix some parentheses warnings
It helps us find a bug in coreml EP:
```
/Users/runner/work/1/s/onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc:93:17: error: using the result of an assignment as a condition without parentheses [-Werror,-Wparentheses]
  if (data_type = ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
```

4. Update cmake to the latest.

5. Remove "--x86" build option from build.py. We can auto-detect cpu architecture from python. So we don't need to ask user to specify it.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/nuget/templates/windowsai.yml [hunk 1]
```diff
@@ -576,31 +576,31 @@ jobs:
         msbuild /p:Platform=x64 Microsoft.AI.MachineLearning.Tests.vcxproj
       workingDirectory: $(Build.SourcesDirectory)\csharp\test\Microsoft.AI.MachineLearning.Tests
 
-  - task: PowerShell@2
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/nuget/templates/windowsai.yml:579]: This file contains a lot of tabs, which should be avoided. So I changed them to spaces.

**Comment by skottmckay** [comment]
> Remove "--x86" build option from build.py. We can auto-detect cpu architecture from python. So we don't need to ask user to specify it.

I have built x86 packages on x64 machines previously to test things (e.g. minimal build size). Won't that remove this ability? I can understand detecting the architecture if no argument is provided though - but shouldn't we allow it to be explicitly specified?

**Comment by snnn** [comment]
> > Remove "--x86" build option from build.py. We can auto-detect cpu architecture from python. So we don't need to ask user to specify it.
> 
> I have built x86 packages on x64 machines previously to test things (e.g. minimal build size). Won't that remove this ability? I can understand detecting the architecture if no argument is provided though - but shouldn't we allow it to be explicitly specified?

Good point. I can keep it. Let it add it back.

File: cmake/onnxruntime_graph.cmake [hunk 2]
```diff
@@ -81,7 +81,9 @@ endif()
 onnxruntime_add_static_library(onnxruntime_graph ${onnxruntime_graph_lib_src})
 add_dependencies(onnxruntime_graph onnx_proto flatbuffers)
 onnxruntime_add_include_to_target(onnxruntime_graph onnxruntime_common onnx onnx_proto ${PROTOBUF_LIB} flatbuffers)
-
+if(NOT MSVC)
+  target_compile_options(onnxruntime_graph PRIVATE "-Wno-parentheses")
```
Comment (askhade) [inline cmake/onnxruntime_graph.cmake:85]: curious: why not for windows? alternatively do we need an if?

Comment (snnn) [reply cmake/onnxruntime_graph.cmake:85]: This compile flag is GCC/Clang only.  Previously we disable this warning globally. Now I scope it to only where ONNX schemas are defined. onnxruntime_graph  contains schemas.  Ideally we should fix all the warnings and remove such flags.

**Comment by snnn** [comment]
The "Nuget WindowsAI Pipeline" failures are not due to this change. It is being addressed separately by someone else.

**Review by liqunfu — APPROVED** [review]
training part LGTM

**Review by gwang-msft — APPROVED** [review]
LGTM

**Pull request merged**